packages/lit-ssr/src/lib/render-lit-html.ts

```diff
@@ -53,30 +53,58 @@ import {
   isElement,
 } from './util/parse5-utils.js';
 
-import {isRenderLightDirective} from 'lit-html/directives/render-light.js';
-import {LitElement} from 'lit-element';
+import {isRenderLightDirective} from 'lit/directives/render-light.js';
+import {LitElement} from 'lit';
 import {LitElementRenderer} from './lit-element-renderer.js';
 import {reflectedAttributeName} from './reflected-attributes.js';
-import {Directive} from 'lit-html/directive';
 
 declare module 'parse5' {
   interface DefaultTreeElement {
     isDefinedCustomElement?: boolean;
   }
 }
 
-patchDirectiveResolve(
-  Directive.prototype,
-  function (this: Directive, values: unknown[]) {
-    const {__part, __attributeIndex} = this;
-    return resolveDirective(
-      __part,
-      this.render(...values),
-      this,
-      __attributeIndex
+/**
+ * Looks for values of type `DirectiveResult` and patches a `Directive` class's
+ * implementation of `_$resolve` to call `render` rather than `update`, per the
+ * SSR contract
+ */
+const patchIfDirective = (value: unknown) => {
+  const directiveCtor = (value as DirectiveResult)?._$litDirective$;
```

Comment:
do you have to check for `undefined` and `null` first? Might be better to check for an object anyway to narrow the set of types you'll do this lookup against.

Reply:
It's optionally-chained...

Reply:
oh... I missed that :)